File: python/structural/bridge/MainBridge.py
from abc import ABC, abstractmethod

class Motor(ABC):
    @abstractmethod
    def getHorsepower(self) -> int:
        pass

class StockMotor(Motor):
    def getHorsepower(self) -> int:
        return 250

class RaceMotor(Motor):
    def getHorsepower(self) -> int:
        return 1000

class Car(ABC):
    def __init__(self, motor: Motor):
        if (isinstance(motor, Motor)):
            self._motor = motor
        else:
            raise ValueError

    def getMotor(self):
        return self._motor
    @abstractmethod
    def getType(self):
        pass

class PassengerCar(Car):
    def getType(self) -> str:
        return "Passenger car with " + str(self._motor.getHorsepower()) + "hp"

class SUV(Car):
    def getType(self) -> str:
        return "SUV with " + str(self._motor.getHorsepower()) + "hp"

def main():
    raceSUV: Car = SUV(RaceMotor())
    print(raceSUV.getType())
    stockCar: Car = PassengerCar(StockMotor())
    print(stockCar.getType())

main()

File: swift5/Sources/swift5/structural/bridge/MainBridge.swift
fileprivate protocol Motor {
	func getHorsepower() -> Int;
}

fileprivate class StockMotor : Motor {
	public func getHorsepower() -> Int {
		return 250;
	}
}

fileprivate class RaceMotor : Motor {
	public func getHorsepower() -> Int {
		return 1000;
	}
}

fileprivate protocol Car {
	func getMotor() -> Motor
	func getType() -> String
}

fileprivate class PassengerCar : Car {
    private var motor: Motor

	init(_ motor: Motor) {
		self.motor = motor;
	}

	public func getMotor() -> Motor {
		return self.motor
	}

	public func getType() -> String {
		return "Passenger car with " + String(self.motor.getHorsepower()) + "hp";
	}
}

fileprivate class SUV : Car {
	private var motor: Motor

	init(_ motor: Motor) {
		self.motor = motor;
	}

	public func getMotor() -> Motor {
		return self.motor
	}

	public func getType() -> String {
		return "SUV with " + String(self.motor.getHorsepower()) + "hp";
	}
}

public class MainBridge {
	public static func main() {
		let raceSUV = SUV(RaceMotor());
		print(raceSUV.getType())
		let stockCar = PassengerCar(StockMotor());
		print(stockCar.getType());
	}
}